packages/client/lib/sync/fetcher/fetcher.ts

```diff
@@ -36,7 +47,12 @@ export interface FetcherOptions {
 export class Fetcher extends Readable {
   public config: Config
 
+  protected chain: Chain
   protected pool: PeerPool
+
+  protected first: BN
+  protected count: BN
```

Comment:
This makes the `Fetcher` only compatible with Block/BlockHeader downloads. We probably want to extend `Fetcher` in the future to also support downloading state types, which is not directly supported if we encapsulate `Fetcher` as this. We will then have to refactor this out later to support this, I am OK with it now (I will probably refactor it out in #1023, where I already moved some methods around).